### PR TITLE
namepet APIs (CLI not included)

### PR DESCRIPTION
**Pull request by complected**
Just a draft to track progress now.

### CODE REVIEW

File: namepet.js [hunk 1]
```diff
@@ -1,36 +1,81 @@
...
+export function make(path = "", verbose = null) {
+  return new Database(path, { verbose })
+}
+
+// https://www.sqlite.org/lang_createtable.html
+export function create(db) {
+  const makestmt = db.prepare(`CREATE TABLE
+                               namepet 
+                               (
+                                 wen TEXT NOT NULL,
```
Comment (nmushegian) [inline namepet.js:17]: Let’s make this a date type that is queryable as such

File: namepet.js [hunk 2]
```diff
@@ -1,36 +1,81 @@
...
+                                 ecr TEXT NOT NULL,
+                                 sig TEXT NOT NULL,
+                                 exp TEXT NOT NULL,
```
Comment (nmushegian) [inline namepet.js:20]: This also is a time type

File: namepet.js [hunk 3]
```diff
@@ -1,36 +1,81 @@
...
+                                 nom TEXT NOT NULL,
+                                 wat TEXT NOT NULL,
+                                 dat TEXT NOT NULL,
+
+                                 PRIMARY KEY (wen, ecr)
```
Comment (nmushegian) [inline namepet.js:25]: Not sure why these would be primary key

File: namepet.js [hunk 4]
```diff
@@ -1,36 +1,81 @@
...
+                                 ON CONFLICT ROLLBACK
```
Comment (nmushegian) [inline namepet.js:26]: Don’t know what this is, is the program incorrect if you delete it?

File: namepet.js [hunk 5]
```diff
@@ -1,36 +1,81 @@
...
+}
+
+// write
+
+export function add(db, sig, exp, nom, wat, dat) {
+  // https://www.sqlite.org/lang_datefunc.html
+  const addstmt = db.prepare(`INSERT INTO 
+                                namepet (wen, ecr, sig, exp, nom, wat, dat) 
+                              VALUES
+                                (strftime('%Y-%m-%dT%H:%M:%fZ', 'now'), ?, ?, ?, ?, ?, ?)
```
Comment (nmushegian) [inline namepet.js:74]: Again should be time types also note there are 2 time types in the record

File: test.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+const sk = ubye(sk_txt);
+const pk = pkey(sk);
+
+test('make and use', async t => {
+  const db = make();
+  t.ok(db, "make - should make");
+  t.ok(create(db), "create - should create");
+  const nom = `vitalik-${Math.floor(Math.random() * 10000)}`;
+  const wat = "https://ethresear.ch";
+  const dat = ":free:vitalik:dai";
```
Comment (nmushegian) [inline test.js:26]: Is this actually a locked path???

File: test.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+const sk_txt = "0x309a49dcda67245af724a05ff081b0c5d24ec0b87a1393d940de6c1dd60f45a3";
+const sk = ubye(sk_txt);
+const pk = pkey(sk);
+
+test('make and use', async t => {
+  const db = make();
+  t.ok(db, "make - should make");
+  t.ok(create(db), "create - should create");
+  const nom = `vitalik-${Math.floor(Math.random() * 10000)}`;
+  const wat = "https://ethresear.ch";
```
Comment (nmushegian) [inline test.js:25]: Let’s not use strings, in fact let’s please explicitly reject strings

File: test.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  const sig = await eip191Sign(msg, sk);
+  t.ok(add(db, pk, sig, exp, nom, wat, dat), "add - should add");
+  t.equal(get(db, nom)[0].nom, nom, "get - should be equal");
+  // verify `ask`, by column
+  [
+    ["nom", nom],
+    ["wat", wat],
+    ["dat", dat],
+    ["exp", exp],
+    ["sig", ustr(sig)],
```
Comment (nmushegian) [inline test.js:38]: We shouldn’t need to use cast functions exported from the library, you should just take the right types to begin with

File: namepet.js [hunk 9]
```diff
@@ -1,36 +1,81 @@
...
+  return new Database(path, { verbose })
+}
+
+// https://www.sqlite.org/lang_createtable.html
+export function create(db) {
+  const makestmt = db.prepare(`CREATE TABLE
+                               namepet 
+                               (
+                                 wen TEXT NOT NULL,
+                                 ecr TEXT NOT NULL,
```
Comment (nmushegian) [inline namepet.js:18]: Some of these have fixed sizes, let’s use the most specific types that make sense

File: package.json [hunk 10]
```diff
@@ -1,8 +1,15 @@
...
+    "type": "module",
     "license": "AGPL-3.0",
     "dependencies": {
         "better-sqlite3": "^7.5.3"
+    },
+    "devDependencies": {
+        "tapzero": "^0.6.1"
+    },
+    "scripts": {
+        "test": "cd coreword && tsc && cd ../ && node test.js"
```
Comment (nmushegian) [inline package.json:13]: You shouldn’t need to do this every time you test

File: namepet.js [hunk 11]
```diff
@@ -1,36 +1,81 @@
...
+                                wat = @txt OR
+                                dat = @txt
+                              `
+  )
+  return askstmt.all({ txt: txt })
+}
+
+// write
+
+export function add(db, sig, exp, nom, wat, dat) {
```
Comment (nmushegian) [inline namepet.js:69]: If you use db as argument style then just put the whole record in second arg like `add(db, [sig exp nom …])`

**Comment by complected** [comment]
Read these comments. And will address them separately in the coming CRs which will be small.

**Pull request closed without merging**